File: 이것이 취업을 위한 코딩 테스트다/BINARY_1.py
import sys

n = int(sys.stdin.readline())
data = list(map(int, sys.stdin.readline().split()))
m = int(sys.stdin.readline())
order = list(map(int, sys.stdin.readline().split()))

def binary(data, target, start, end):
    if start>end:
        return -1
    mid = (start+end)//2
    if data[mid] == target:
        return mid+1
    elif data[mid]>target:
        return binary(data, target, start, mid-1)    
    else:
        return binary(data, target, mid+1, end)
    
for i in range(m):
    result = binary(data, order[i], 0, n-1)
    if result==-1:
        print("no", end=" ")
        continue
    print("yes", end=" ")

File: 이것이 취업을 위한 코딩 테스트다/java/Binary1.java
package algorithm_study;

import java.util.*;

public class Binary1 {
	
	public static Scanner sc;
	public static int n;
	public static int[] parts;
	public static int m;

	
	public static void binary(int target, int start, int end) {
		
		int mid = (start + end) / 2;
		if (parts[mid] == target) {
			System.out.print("yes ");
			return;
		}
		if (start > end) {
			System.out.print("no ");
			return;
		}
		
		if (target > parts[mid]) {
			start = mid + 1;
			binary(target, start, end);
		}
		else {
			end = mid - 1;
			binary(target, start, end);
		}
			
	}

	public static void main(String[] args) {
		
		sc = new Scanner(System.in);
		n = sc.nextInt();

		parts = new int[n];
		for (int i = 0 ; i < n ; i++)
			parts[i] = sc.nextInt();
		sc.nextLine();
		m = sc.nextInt();
		Arrays.sort(parts);

		for (int i = 0 ; i < m ; i++) {
			int target = sc.nextInt();
			binary(target, 0, n - 1);		
		}
		
	}

}
